packages/taquito-sapling/src/types.ts

```diff
@@ -23,11 +36,129 @@ export interface Ciphertext {
   nonceOut: Buffer;
 }
 
-export interface SaplingTransaction {
-  spendDescriptions: SpendDescription[];
-  outputDescriptions: OutputDescription[];
+export interface SaplingTransactionInput {
+  cv: Buffer;
+  nf: Buffer;
+  rk: Buffer;
+  proof: Buffer;
   signature: Buffer;
+}
+
+export interface SaplingSpendDescription extends SaplingTransactionInput {
+  rt: Buffer;
+}
+
+export interface SaplingTransactionOutput {
+  cm: Buffer;
+  proof: Buffer;
+  ciphertext: Ciphertext;
+}
+
+export interface ParametersSaplingTransaction {
+  to: string;
+  amount: number;
+  memo?: string;
+  mutez?: boolean;
+}
+
+export type ParametersUnshieldedTransaction = Omit<ParametersSaplingTransaction, 'memo'>;
+
+export interface ParametersSpendProof {
+  saplingContext: number;
+  sk: Uint8Array;
+  address: Uint8Array;
+  rcm: Uint8Array;
+  ar: Buffer;
+  amount: string;
+  root: string;
+  witness: string;
+}
+
+export interface ParametersSpendSig {
+  sk: Uint8Array;
+  ar: Buffer;
+  unsignedSpendDescription: Omit<SaplingSpendDescription, 'signature'>;
+  hash: Uint8Array;
+}
+
+export interface ParametersOutputDescription {
+  saplingContext: number;
+  address: Uint8Array;
+  amount: string;
+  memo: string;
+  rcm: Buffer;
+  ovk?: Buffer;
+}
+
+export interface ParametersOutputProof {
+  saplingContext: number;
+  esk: Buffer;
+  address: Uint8Array;
+  rcm: Uint8Array;
+  amount: string;
+}
+
+export interface ParametersBindingSig {
+  saplingContext: number;
+  inputs: SaplingTransactionInput[];
+  outputs: SaplingTransactionOutput[];
   balance: BigNumber;
-  root: Buffer;
   boundData: Buffer;
 }
+
+export interface ParametersCiphertext {
+  address: Uint8Array;
+  esk: Buffer;
+  diversifier: Buffer;
+  ock: Uint8Array;
+  amount: string;
+  rcm: Buffer;
+  memo: string;
+}
+
+export interface SaplingTransactionPlaintext {
+  diversifier: Buffer;
+  amount: string;
+  rcm: Buffer;
+  memoSize: number;
+  memo: string;
+}
+
+export interface Input {
+  value: Uint8Array;
+  memo: Uint8Array;
+  paymentAddress: Uint8Array;
+  rcm: Uint8Array;
+  position: number;
+}
+
+export type SaplingContractId =
+  | {
+      contractAddress?: never;
+      saplingId: string;
+    }
+  | {
+      contractAddress: string;
+      saplingId?: never;
+    };
+
+export interface ChosenSpendableInputs {
+  inputsToSpend: Input[];
+  sumSelectedInputs: BigNumber;
+}
+
+export interface SaplingTransactionParamsTxBuilder {
+  root: string;
+  saplingTransactionParams: {
+    to: string;
+    amount: string;
+    memo: string;
+  }[];
```

Comment:
maybe pull this type into another type? this same object is passed in as the param type and could use an interface to save on space and clutter

Reply:
I am not sure if I understand, do you refer to using `ParametersSaplingTransaction` instead? The properties type are not exactly the same.